ogre/src/OgreLidarVisual.cc

```diff
@@ -0,0 +1,301 @@
+  this->dataPtr->receivedMsg = true;
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Update()
+{
+  if (!this->dataPtr->receivedMsg || this->dataPtr->lidarPoints.size() == 0)
+  {
+    ignerr << "Message not received. Exiting update function"
+           << std::endl;
+    return;
+  }
+
+  this->dataPtr->receivedMsg = false;
+  double horizontalAngle = this->minHorizontalAngle;
+  double verticalAngle = this->minVerticalAngle;
+
+  if (this->horizontalCount > 1)
+  {
+    this->horizontalAngleStep =
+        (this->maxHorizontalAngle - this->minHorizontalAngle) /
+              (this->horizontalCount - 1);
+  }
+  if (this->verticalCount > 1)
+  {
+    this->verticalAngleStep =
+        (this->maxVerticalAngle - this->minVerticalAngle) /
+              (this->verticalCount - 1);
+  }
+
+  if (this->dataPtr->lidarPoints.size() !=
+                  this->verticalCount * this->horizontalCount)
+  {
+    ignerr << "Size of laser message inconsistent with ray count"
+           << std::endl;
+    return;
+  }
+
+  // Process each point from message
+  // Every line segment, and every triangle is saved separately,
+  // as a pointer to a DynamicLine
+  for (unsigned int j = 0; j < this->verticalCount; ++j)
+  {
+    horizontalAngle = this->minHorizontalAngle;
+
+    if (j+1 > this->dataPtr->rayLines.size())
+    {
+      // Ray Strips fill in between the line areas that intersect an object
+      std::shared_ptr<OgreDynamicLines> line =
+                  std::shared_ptr<OgreDynamicLines>(
+                              new OgreDynamicLines(MT_TRIANGLE_STRIP));
+
+      line->setMaterial("Lidar/BlueStrips");
+      std::shared_ptr<Ogre::MovableObject> mv =
+                std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+
+      this->Node()->attachObject(mv.get());
+      this->dataPtr->rayStrips.push_back(line);
+
+      line = std::shared_ptr<OgreDynamicLines>(
+                  new OgreDynamicLines(MT_TRIANGLE_STRIP));
+
+      line->setMaterial("Lidar/LightBlueStrips");
+      mv = std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+      this->Node()->attachObject(mv.get());
+      this->dataPtr->noHitRayStrips.push_back(line);
+
+      line = std::shared_ptr<OgreDynamicLines>(
+                  new OgreDynamicLines(MT_TRIANGLE_FAN));
+
+      line->setMaterial("Lidar/TransBlack");
+      mv = std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+      this->Node()->attachObject(mv.get());
+      this->dataPtr->deadZoneRayFans.push_back(line);
+      this->dataPtr->deadZoneRayFans[j]->AddPoint(
+                  ignition::math::Vector3d::Zero);
+
+
+      line = std::shared_ptr<OgreDynamicLines>(
+                  new OgreDynamicLines(MT_LINE_LIST));
+
+      line->setMaterial("Lidar/BlueRay");
+      mv = std::dynamic_pointer_cast<Ogre::MovableObject>(line);
+      this->Node()->attachObject(mv.get());
+      this->dataPtr->rayLines.push_back(line);
+
+      this->SetVisibilityFlags(0x0FFFFFFF);
```

Comment:
let's remove this line for setting the visibility flags here. I think this should be done later by ign-gazebo.